service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/AllocationDecisionControllerAboutToStartTest.java

```diff
@@ -58,6 +58,104 @@ void shouldAddYesToMissingAllocationDecision() throws Exception {
+            .judgeLevelRadio("No")
+            .allocationProposalPresent("Yes")
+            .build();
+
+        CaseData caseData = mapper.convertValue(response.getData(), CaseData.class);
+        AllocationDecision actualAllocationDecision = caseData.getAllocationDecision();
+        assertThat(actualAllocationDecision)
+            .isEqualTo(expectedDecision);
+    }
+
+    @Test
+    void shouldAddNullToJudgeLevelRadio() throws Exception {
+
+        AllocationProposal allocationProposal = AllocationProposal.builder()
+            .proposal("proposal")
+            .proposalReason("reason")
+            .build();
+
+        CallbackRequest request = CallbackRequest.builder().caseDetails(CaseDetails.builder()
+            .data(ImmutableMap.<String, Object>builder()
+                .put("allocationProposal", allocationProposal)
+                .build()).build())
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse response = callbackResponse(request);
+
+        AllocationDecision expectedDecision = AllocationDecision.builder()
+            .judgeLevelRadio(null)
+            .allocationProposalPresent("Yes")
+            .build();
+
+        CaseData caseData = mapper.convertValue(response.getData(), CaseData.class);
+        AllocationDecision actualAllocationDecision = caseData.getAllocationDecision();
+        assertThat(actualAllocationDecision)
+            .isEqualTo(expectedDecision);
+    }
+
+    @Test
+    void shouldAddYesToJudgeLevelRadio() throws Exception {
+
+        AllocationProposal allocationProposal = AllocationProposal.builder()
+            .proposal("proposal")
+            .proposalReason("reason")
+            .build();
+
+        AllocationDecision allocationDecision = AllocationDecision.builder()
+            .proposal("proposal")
+            .proposalReason("reason")
+            .build();
+
+        CallbackRequest request = CallbackRequest.builder().caseDetails(CaseDetails.builder()
+            .data(ImmutableMap.<String, Object>builder()
+                .put("allocationProposal", allocationProposal)
+                .put("allocationDecision", allocationDecision)
+                .build()).build())
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse response = callbackResponse(request);
+
+        AllocationDecision expectedDecision = AllocationDecision.builder()
+            .judgeLevelRadio("Yes")
+            .proposal("proposal")
+            .proposalReason("reason")
+            .allocationProposalPresent("Yes")
+            .build();
+
+        CaseData caseData = mapper.convertValue(response.getData(), CaseData.class);
+        AllocationDecision actualAllocationDecision = caseData.getAllocationDecision();
+        assertThat(actualAllocationDecision)
+            .isEqualTo(expectedDecision);
+    }
+
+    @Test
+    void shouldAddNoToJudgeLevelRadio() throws Exception {
+
+        AllocationProposal allocationProposal = AllocationProposal.builder()
```

Comment:
we're building lots of the same objects, maybe extract to private test methods to shorten the test line length?